src/app/frontend/common/autorefresh/poll_service.js

```diff
@@ -0,0 +1,208 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Service class for polling.
+ * @final
+ */
+export class PollService {
+  /**
+   * @param $interval {!angular.$interval}
+   * @param $q {!angular.$q}
+   * @ngInject
+   */
+  constructor($interval, $q) {
+    /**
+     * @type {!angular.$interval}
+     * @private
+     */
+    this.intervalService = $interval;
+
+    /**
+     * @type {!angular.$q}
+     * @private
+     */
+    this.q_ = $q;
+
+    /**
+     * @type {!Array<PollOptions>}
+     * @private
+     */
+    this.optionsList;
+
+    /**
+     * @type {!PollOptions}
+     */
+    this.currentOptions;
+  }
+
+  /**
+   * Starts polling and returns polling promise.
+   * @param resource {!angular.$resource}
+   * @param params {angular.Resource.ParamsOrCallback=}
+   * @param delay {number}
+   * @returns {!angular.$q.Promise}
+   */
+  poll(resource, params, delay) {
+    this.optionsList = this.optionsList || new Array();
+    if (!this.isResourceExist(resource)) {
+      this.currentOptions = new PollOptions(resource, params, delay);
+      this.optionsList.push(this.currentOptions);
+    }
+
+    return this.start();
+  }
+
+  /**
+   * Stops the current polling interval.
+   * @export
+   */
+  stop() {
+    if (!this.currentOptions) {
+      return;
+    }
+    this.cancelInterval(this.currentOptions);
+  }
+
+  /**
+   * Starts polling interval for the current options.
+   * @export
+   * @returns {!angular.$q.Promise}
+   */
+  start() {
+    if (!this.currentOptions) {
+      return this.q_.defer().promise;
+    }
+    this.stop();
+    return this.createInterval(this.currentOptions);
+  }
+
+  /**
+   * Stops polling interval for the current options.
+   * @export
+   */
+  stopAll() {
+    this.optionsList.map((options) => {
+      this.cancelInterval(options);
+    });
+  }
+
+  /**
+   * Starts polling intervals for the all registered options.
+   * @export
+   */
+  startAll() {
+    this.stopAll();
+    this.optionsList.map((options) => {
+      this.createInterval(options);
+    });
+  }
+
+  /**
+   * Creates polling interval.
+   * @param options {!PollOptions}
+   * @returns {!angular.$q.Promise}
+   * @private
+   */
+  createInterval(options) {
+    if (!options.deferred) {
+      options.deferred = this.q_.defer();
+    }
+
+    options.interval = this.intervalService(() => {
+      let promise = options.resource_.get(options.params).$promise;
+      promise.then(
+          (obj) => {
+            options.deferred.notify(obj);
+          },
+          (err) => {
+            options.deferred.reject(err);
+            // Resource can't retrieve any record. It is not necessary to continue polling anymore.
+            this.stop();
```

Comment:
Correct. 
In this case the failing options is not necessarily current one. I will modify it to stop and remove the failing one instead of current one.